day6_2018.py

```diff
@@ -0,0 +1,156 @@
+with open("input6.txt", "r") as file:
+    inputs = file.read().split("\n")
+from collections import namedtuple
+
+inputs = [x.split(", ") for x in inputs]
+Point = namedtuple("Point", ["x", "y"])
+
+
+def get_coords_from_input(inputs):
+    x_coords = [int(coord[0]) for coord in inputs]
+    y_coords = [int(coord[1]) for coord in inputs]
+    return x_coords, y_coords
+
+
+def get_points_for_manhattan_distance(row_idx, column_idx, x_coord, y_coord):
+    return Point(x=x_coord, y=y_coord), Point(x=row_idx, y=column_idx)
+
+
+def manhattan_distance(first_and_second_points):
+    first_point, second_point = first_and_second_points[0], first_and_second_points[1]
+    return abs(first_point.x - second_point.x) + abs(first_point.y - second_point.y)
+
+
+def coords_extremes(x_coords, y_coords):
+    return min(x_coords), max(x_coords), min(y_coords), max(y_coords)
+
+
+def get_distance_to_all_points(x_coords, y_coords):
+    all_distances, all_x_coords, all_y_coords = [], [], []
+    for x_coord, y_coord in zip(x_coords, y_coords):
+        distance = []
+        for row_idx in range(max(x_coords)):
+            for column_idx in range(max(y_coords)):
+                distance.append(
+                    manhattan_distance(
+                        get_points_for_manhattan_distance(
+                            row_idx, column_idx, x_coord, y_coord
+                        )
+                    )
```

Comment:
O ile jestem fanem wielu małych funkcji, to w tym wypadku, wydaje mi się, że funkcja `get_points_for_manhattan_distance` zaciemnia kwestię i później niepotrzebnie zwraca krotkę, po której znów indeksujemy.

Proponowałbym usunąć tę funkcję i zrobić coś takiego:

```suggestion
                    manhattan_distance(first_point=Point(x=x_coord, y=y_coord), second_point=Point(x=row_idx, y=column_idx))
```

Ofc sformatować. Do tego w funkcji `manhattan_distances` zmienić argument `first_and_second_points` na dwa i dzięki temu nie będzie tam już tego pakudnego indexowania :stuck_out_tongue: